packages/EasyCore/src/Bridge/Symfony/Validator/Constraints/AbnValidator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Bridge\Symfony\Validator\Constraints;
+
+use EoneoPay\Utils\Exceptions\UnexpectedTypeException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * Validates an Australian Business Number.
+ *
+ * @link https://abr.business.gov.au/Help/AbnFormat
+ */
+final class AbnValidator extends ConstraintValidator
+{
+    /**
+     * @var int
+     */
+    private const ABN_LENGTH = 11;
+
+    /**
+     * @var int
+     */
+    private const MODULUS = 89;
+
+    /**
+     * @var int[]
+     */
+    private const WEIGHTS = [10, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
+
+    /**
+     * Validates an Australian Business Number.
+     *
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
```

Comment:
Please let's make sure we move away from this `EoneoPay\Utils` package as it will be deprecated soon. And I think that we should use the `UnexpectedValueException` from Symfony if possible, otherwise let's just create our own exception within easy-core for now.

Reply:
agree

Reply:
Done.